.github/workflows/push-pr-unit-tests.yml

```diff
@@ -0,0 +1,17 @@
+name: unit tests
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  push-pr-unit-tests:
```

Comment:
Isn't the trick here to name this `build` so it still fits the repo settings? This way we shouldn't need to reconfigure the repo and rebase old PRs?